filers/default/src/main/java/org/commonjava/indy/filer/def/DefaultGalleyStorageProvider.java

```diff
@@ -135,6 +135,30 @@ public DefaultGalleyStorageProvider( final File storageRoot, final File nfsStora
     @PostConstruct
     public void setup()
     {
+        getSpecialPathManager();
+
+        setupTransferDecoratorPipeline();
+
+        final File storeRoot = config.getStorageRootDirectory();
+
+        // Apply partyline gloable lock manager if in cluster Env
+        GlobalLockManager globalLockManager = null;
+        if ( indyConfiguration.isClusterEnabled() )
+        {
+            logger.info( "Enable ISPN transactional GLM via {}", partylineGLMCache.getName() );
+            globalLockManager = new InfinispanTransactionalGLM( partylineGLMCache.getCache() );
+        }
+        cacheProviderFactory = new PartyLineCacheProviderFactory( storeRoot, deleteExecutor, globalLockManager );
+
+        // TODO: Tie this into a config file!
+        transportManagerConfig = new TransportManagerConfig();
+
+        // Little hacky here. Because there are loop dependency between SpecialPathManager, TransferDecorator and
```

Comment:
I put a comment in the other PR...this is tricky for sure. I wonder if we could use the PathGenerator directly from SpecialPathManagerImpl instead, and break the dep cycle between SpecialPathManager, the TransferDecorators, and CacheProvider?